src/t_zset.c

```diff
@@ -72,12 +72,41 @@ void zsetConvertAndExpand(robj *zobj, int encoding, unsigned long cap);
 zskiplistNode *zslGetElementByRankFromNode(zskiplistNode *start_node, int start_level, unsigned long rank);
 zskiplistNode *zslGetElementByRank(zskiplist *zsl, unsigned long rank);
 
+static inline unsigned long zslGetNodeSpanAtHeight(zskiplistNode *x, int height) {
+    return height > 0 ? x->level[height].span : x->level[height].forward ? 1
+                                                                         : 0;
+}
+
+static inline void zslSetNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long span) {
+    if (height > 0)
+        x->level[height].span = span;
+}
+
+static inline void zslIncrNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long incr) {
+    if (height > 0)
+        x->level[height].span += incr;
+}
+
+static inline void zslDecrNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long decr) {
+    if (height > 0)
+        x->level[height].span -= decr;
+}
+
+static inline unsigned long zslGetNodeHeight(zskiplistNode *x) {
+    return x->level[0].span;
+}
+
+static inline void zslSetNodeHeight(zskiplistNode *x, int height) {
+    x->level[0].span = height;
+}
+
 /* Create a skiplist node with the specified number of levels.
  * The SDS string 'ele' is referenced by the node after the call. */
 zskiplistNode *zslCreateNode(int level, double score, sds ele) {
     zskiplistNode *zn = zmalloc(sizeof(*zn) + level * sizeof(struct zskiplistLevel));
     zn->score = score;
     zn->ele = ele;
+    zslSetNodeHeight(zn, level - 1);
```

Comment:
It looks odd that we set height to `level - 1`. Isn't the height the number of levels in a node?

Is `level` here the number of levels here? Maybe we should rename the level parameter to height?

Reply:
So yes. height is the number of levels in the node. I can make this change

Reply:
done